### PR TITLE
Add DLQ Metadata

### PR DESCRIPTION
**Pull request by georgeteo**
Add protobuf backed DLQ metadata

### CODE REVIEW

File: internal/consumer/dlqMetadata.proto [hunk 1]
```diff
@@ -0,0 +1,10 @@
+syntax = "proto3";
+
+message dlqMetadata {
+    int64 retry_count = 1;
+    string topic = 2;
+    int32 partition = 3;
+    int64 offset = 4;
+    int64 timestamp_ns = 5;
```
Comment (venkat1109) [inline internal/consumer/dlqMetadata.proto]: Go naming style: timestamp_nanos
https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: internal/consumer/dlqMetadata.pb.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
```
Comment (venkat1109) [inline internal/consumer/dlqMetadata.pb.go]: how is this file generated ? Can you add a codgen target to the Makefile that can be used to update the generated file when the proto file changes ?

File: internal/consumer/dlqMetadata.proto (hunk 1 above)
Comment (georgeteo) [reply internal/consumer/dlqMetadata.proto]: I prefer timestamp_ns and naming conventions allows for an exception for protobuf generated code.

File: internal/consumer/dlqMetadata.pb.go (hunk 2 above)
Comment (georgeteo) [reply internal/consumer/dlqMetadata.pb.go]: It requires the person updating to install the protobuf compiler binary distributed by Google. I think we can leave it out for now.

**Pull request merged**